docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,434 @@
+```
+
+##### `/device/{name}` with managementMode: fleet
+
+When managing a device through a fleet, flightctl is responsible
+for maintaining the device spec in sync with the fleet spec; in this case
+write access is only possible for the modification of labels and other metadata
+through this API, `spec.managementMode` can also be set to `manual`.
+
+A device can be moved into an manual management state by flipping the `spec.managementMode`
+field. This is useful to allow temporary drift for debugging specific devices. As soon
+as this field is flipped back to `fleet`, the device controller will roll back the device
+configuration to the fleet mandated configuration and remove any modifications made to the
+device spec.
+
+Devices set to `spec.managementMode: manual`, but with a `metadata.owner` fleet , must signal
+a device status condition to indicate that the device is not in sync with the fleet because of
+this.
+
+```yaml
+  - lastTransitionTime: "2024-04-30T15:06:20Z"
+    message: This device is not in sync with the fleet
+    reason: Manually Managed
+    status: False
+    type: Management
+```
+
+When a fleet owned device is moved back to fleet management the status condition should be removed.
+
+##### `/device/{name}` with managementMode: manual
+
+Devices in manual management mode are not managed by the fleet controller, the device
+controller will not attempt to reconcile the device spec with the fleet spec.
+
+In this mode, the device.spec is fully writtable through this API.
+
+If the device is selected by a fleet, the `spec.managementMode` can be flipped to `fleet`, and the
+device controller will start managing the device again.
+
+While devices are managed by fleets, all floating tags are resolved to stable hashes, but
+when a device is managed manually, the floating tags can be used, see the next sub-section
+for additional details.
+
+###### Expected behaviour regarding external references with floating tags
+
+Changes on external references is monitored and acted on (git branches,
+device images, secrets), the device controller should re-evaluate the device
+spec and apply the changes to the rendered version which will be exposed
+on the `/device/{name}/rendered` endpoint.
+
+Several mechanisms can be used to track external references:
+* Polling
+* Webhooks
+
+#### `/device/{name}/rendered` (device side)
+This endpoint is used by the device agent to retrieve the rendered device spec, this endpoint is
+read-only, each device can only access it's own device rendered spec based on the device
+certificate identification, an administrator can access any device rendered spec for debug purposes.
+
+The agent can pass two parameters: `version`and `owner` if those haven't changed the API will return
+204 No Content, if the version or fleet has changed the API will return the rendered spec.
+
+Change proposed:
+* The parameter name `fleet` is changed to `owner` to be consistent with the metadata field.
```

Comment:
Since the agent treats both strings as opaque, FC should maybe provide a single string and call it something like "latestSeenID."

Reply:
Right now it's owner plus version.  I'm suggesting to just have some token/identifier for the specification.  LatestSpecificationID?

Reply:
Yes, resourceversion sounds right